tests/reduct/ContinReductUTest.cxxtest

```diff
@@ -31,209 +31,293 @@
 using namespace std;
 using namespace opencog::combo;
 using namespace opencog::reduct;
+using namespace opencog;
+
+#define al _as.add_link
+#define an _as.add_node
+
+class ContinReductUTest : public CxxTest::TestSuite
+{
+private:
+	AtomSpace _as;
+	Handle ZERO, X, XPLUS0;
+
+public:
+	ContinReductUTest()
+	{
+		ZERO = an(NUMBER_NODE, "0");
+		X = an(NUMBER_NODE, "5");
+		XPLUS0 = al(PLUS_LINK, {X, ZERO});
+	}
+
+private:
+	// test if the combo program of string tr_str is shorter (or equal
+	// if strict is false) to the result once reduced, and preserves
+	// semantics (compare contin table)
+	void perform_test_reduce(const string &tr_str, bool strict = true)
+	{
+		stringstream ss(tr_str);
+		combo_tree tr;
+		ss >> tr;
+
+		type_tree tt = infer_type_tree(tr);
+
+		ITable rnt(tt);
+
+		int size_before, size_after;
+		int reduct_effort = 2;
+
+		try {
+			OTable ct1(tr, rnt);
+			// std::cout << "ct1:" << std::endl << ct1 << std::endl;;
+			size_before = tr.size();
+			std::cout << "before: " << tr << std::endl;
+
+			contin_reduce(tr, reduct_effort, vertex_set());
+			std::cout << "after:  " << tr << std::endl;
+			size_after = tr.size();
+			OTable ct2(tr, rnt);
+			// std::cout << "ct2:" << std::endl << ct2 << std::endl;;
+
+			if (strict) {
+				TS_ASSERT_LESS_THAN(size_after, size_before);
+			} else {
+				TS_ASSERT_LESS_THAN_EQUALS(size_after, size_before);
+			}
+			TS_ASSERT_EQUALS(ct1, ct2);
+		}
+		catch (EvalException &e) {
+			std::cout << "An expection has been raised during the test (could be normal)" << std::endl;
+			TS_ASSERT(true);
+		}
+	}
+
+	void perform_test_reduce(Handle &h, bool strict = true)
+	{
+		int size_before, size_after;
+		int reduct_effort = 2;
+
+		AtomeseToCombo to_combo;
+		combo_tree tr_before = to_combo(h).first;
+
+		type_tree tt = infer_type_tree(tr_before);
+		ITable rnt(tt);
+
+		contin_reduce(h, reduct_effort, HandleSet());
+
+		combo_tree tr_after = to_combo(h).first;
+
+		try {
+			OTable ct1(tr_before, rnt);
```

Comment:
And we probably want to overload OTable to support atomese, but again, let's worry about later.